core/math/a_star.cpp

```diff
@@ -190,8 +190,7 @@ bool AStar::_solve(Point *begin_point, Point *end_point) {
 
 		Point *n = begin_point->neighbours[i];
 		n->prev_point = begin_point;
-		n->distance = _compute_cost(n->id, begin_point->id);
-		n->distance *= n->weight_scale;
+		n->distance = _compute_cost(begin_point->id, n->id) * n->weight_scale;
```

Comment:
I swapped the arguments here, since `_compute_cost` takes `(p_from_id, p_to_id)`, and we are obviously moving from the initial point to one of the neighbouring points.